Mathlib/Order/RelSeries.lean

```diff
@@ -561,6 +561,40 @@ lemma smash_succ_natAdd {p q : RelSeries r} (h : p.last = q.head) (i : Fin q.len
     (smash p q h).last = q.last := by
   delta smash last; aesop
 
+/-- Given the series `a₀ -r→ … -r→ aᵢ -r→ … -r→ aₙ`, the series `a₀ -r→ … -r→ aᵢ`. -/
+@[simps! length]
+def take {r : Rel α α} (p : RelSeries r) (i : Fin (p.length + 1)) : RelSeries r where
+  length := i
+  toFun := fun ⟨j, h⟩ => p.toFun ⟨j, by omega⟩
```

Comment:
I thought a little about
```suggestion
  toFun := p.toFun ∘ Fin.castLE (Nat.succ_le_of_lt i.2)
```
which maybe helps somewhere by making `castLE` lemmas available; but I can't think of an analogous spelling for `drop`, so this probably isn't helpful.